seqr/views/apis/locus_list_api.py

```diff
@@ -0,0 +1,233 @@
+            BaseProjectGeneList.objects.filter(project=xbrowse_project, gene_list=xbrowse_gene_list).delete()
+
+    return create_json_response({
+        'locusListGuids': [locus_list['locusListGuid'] for locus_list in get_sorted_project_locus_lists(project, request.user)],
+    })
+
+
+def get_project_locus_list_models(project):
+    return get_objects_for_group(project.can_view_group, CAN_VIEW, LocusList)
+
+
+def get_sorted_project_locus_lists(project, user):
+    result = get_json_for_locus_lists(get_project_locus_list_models(project), user)
+    return sorted(result, key=lambda locus_list: locus_list['name'])
+
+
+def add_locus_list_user_permissions(locus_list):
+    assign_perm(user_or_group=locus_list.created_by, perm=IS_OWNER, obj=locus_list)
+    assign_perm(user_or_group=locus_list.created_by, perm=CAN_EDIT, obj=locus_list)
+    assign_perm(user_or_group=locus_list.created_by, perm=CAN_VIEW, obj=locus_list)
+
+
+def _update_locus_list(locus_list, request_json, user):
+    requested_items = (request_json.get('parsedItems') or {}).get('items') or []
+
+    existing_gene_ids = set()
+    new_gene_symbols = set()
+    new_gene_ids = set()
+    existing_interval_guids = set()
+    new_intervals = []
+    invalid_items = []
+    for item in requested_items:
+        if item.get('locusListIntervalGuid'):
+            existing_interval_guids.add(item.get('locusListIntervalGuid'))
+        elif item.get('geneId'):
+            if item.get('symbol'):
+                existing_gene_ids.add(item.get('geneId'))
+            else:
+                new_gene_ids.add(item.get('geneId'))
+        elif item.get('symbol'):
+            new_gene_symbols.add(item.get('symbol'))
+        else:
+            try:
+                item['start'] = int(item['start'])
+                item['end'] = int(item['end'])
+                if item['start'] > item['end']:
+                    raise ValueError
+                get_xpos(item['chrom'], int(item['start']))
+                new_intervals.append(item)
+            except (KeyError, ValueError):
+                invalid_items.append('chr{chrom}:{start}-{end}'.format(
+                    chrom=item.get('chrom', '?'), start=item.get('start', '?'), end=item.get('end', '?')
+                ))
+
+    gene_symbols_to_ids = get_gene_symbols_to_gene_ids(new_gene_symbols)
+    invalid_items += [symbol for symbol, gene_id in gene_symbols_to_ids.items() if not gene_id]
+    new_genes = get_genes([gene_id for gene_id in gene_symbols_to_ids.values() if gene_id] + list(new_gene_ids))
+    invalid_items += [gene_id for gene_id, gene in new_genes.items() if not gene]
+    new_genes = {gene_id: gene for gene_id, gene in new_genes.items() if gene}
+
+    if invalid_items and not request_json.get('ignoreInvalidItems'):
+        return None, None, invalid_items, 'This list contains invalid genes/ intervals. Update them, or select the "Ignore invalid genes and intervals" checkbox to ignore.'
```

Comment:
could the invalid items be appended to this error message, so the error box shows one message instead of 2?